rest/route/distro.go

```diff
@@ -440,6 +444,121 @@ func (h *distroAMIHandler) Run(ctx context.Context) gimlet.Responder {
 		"no settings available for region '%s' for distro '%s'", h.region, h.distroID))
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/distros/settings modifies provider settings across all distros for the given region
+
+type modifyDistrosSettingsHandler struct {
+	//settings *cloud.EC2ProviderSettings
+	settings *birch.Document
+	region   string
+
+	sc data.Connector
+}
+
+func makeModifyDistrosSettings(sc data.Connector) gimlet.RouteHandler {
+	return &modifyDistrosSettingsHandler{
+		settings: &birch.Document{},
+		sc:       sc,
+	}
+}
+
+func (h *modifyDistrosSettingsHandler) Factory() gimlet.RouteHandler {
+	return &modifyDistrosSettingsHandler{
+		settings: &birch.Document{},
+		sc:       h.sc,
+	}
+}
+
+func (h *modifyDistrosSettingsHandler) Parse(ctx context.Context, r *http.Request) error {
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	b, err := ioutil.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+	if err = json.Unmarshal(b, h.settings); err != nil {
+		return errors.Wrap(err, "API error while unmarshalling JSON")
+	}
+
+	var ok bool
+	h.region, ok = h.settings.Lookup("region").StringValueOK()
+	if !ok || h.region == "" {
+		return gimlet.ErrorResponse{
+			Message:    "region must be explicitly defined",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	return nil
+}
+
+func (h *modifyDistrosSettingsHandler) Run(ctx context.Context) gimlet.Responder {
+	u := MustHaveUser(ctx)
+	allDistros, err := h.sc.FindAllDistros()
```

Comment:
I don't really know the use cases, but are there use cases for doing this by more granularity than all ec2 distros? Like maybe those with a certain manager type or maybe a regex for the distro ID?

Reply:
I thought about this, but at this point I'm skeptical we'll need this for non ec2 distros (in the short term we need it for region cleanup). Also the validation is all region specific which doesn't impact other distros. 

I figure if we have another use case in the future to extend this we can.